.github/workflows/cmake.yml

```diff
@@ -86,62 +69,89 @@ jobs:
     # TODO: ccache
     # https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/
     # https://github.com/cristianadam/HelloWorld/blob/master/.github/workflows/build_cmake.yml
+    - name: Create build environment with debug macros
+      working-directory: ${{ github.workspace }}
+      # Build the entire project with debug macros turned on, to prevent code rot.
+      # However, it is unweildy and slow to run tests with them enabled, so we will build twice.
+      run: |
+        mkdir -p Build
+        cd Build
+        cmake .. -GNinja -DBUILD_LAGOM=ON -DENABLE_ALL_THE_DEBUG_MACROS=ON -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
+      if: ${{ matrix.debug-macros == 'ALL_DEBUG' }}
     - name: Create build environment
       working-directory: ${{ github.workspace }}
       # Note that this needs to run *even if* the Toolchain was built,
       # in order to set options like BUILD_LAGOM.
       run: |
         mkdir -p Build
         cd Build
-        cmake .. -GNinja -DBUILD_LAGOM=ON -DENABLE_ALL_THE_DEBUG_MACROS=ON -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
+        cmake .. -GNinja -DBUILD_LAGOM=ON -DCMAKE_C_COMPILER=gcc-10 -DCMAKE_CXX_COMPILER=g++-10
+      if: ${{ matrix.debug-macros == 'NORMAL_DEBUG' }}
 
-    # === ACTUALLY BUILD AND TEST ===
+    # === ACTUALLY BUILD ===
 
     - name: Build Serenity and Tests
       working-directory: ${{ github.workspace }}/Build
       run: cmake --build .
     - name: Lint (Phase 2/2)
       working-directory: ${{ github.workspace }}/Meta
       run: ./check-symbols.sh
-    - name: Run CMake tests
-      working-directory: ${{ github.workspace }}/Build
-      run: CTEST_OUTPUT_ON_FAILURE=1 ninja test || ${{ matrix.allow-test-failure }}
-      timeout-minutes: 2
-    - name: Run JS tests
-      working-directory: ${{ github.workspace }}/Build/Meta/Lagom
-      run: DISABLE_DBG_OUTPUT=1 ./test-js || ${{ matrix.allow-test-failure }}
-    - name: Run LibCompress tests
-      working-directory: ${{ github.workspace }}/Build/Meta/Lagom
-      run: ./test-compress
-
-    # === NOTIFICATIONS ===
 
-    - name: Dump event info
-      if: always()
-      # Usually unnecessary, but insanely useful if IRC notifications fail.
-      run: |
-        cat <<"EOF"
-        ${{ toJSON(github.event) }}
-        EOF
-    - name: Generate IRC message
-      if: matrix.should-notify-irc == true && !cancelled()
+    - name: Create Serenity Rootfs
+      if: ${{ matrix.debug-macros == 'NORMAL_DEBUG'}}
+      working-directory: ${{ github.workspace }}/Build
+      # See Documentation/RunningTests.md for copy-pasteable version of this addition
+      #    (Also please keep them in sync :) )
```

Comment:
Well, boot mode is a global string in SystemServer, and all the services check whether their boot mode is contained w/in the boot_mode string before trying to start. So to do what you're suggesting we'd need to add "self-test" to enough services that the tests could actually run, but it would be possible yes. I'll look into that one. I never did like echoing a bunch of stuff into the file in the ci script.

Reply:
All right I pushed a "self-test" mode that's an alias for text mode. Bit gnarly that setting up the framebuffer for graphical is handled in the kernel still but whatevs, it works as written.